monkey/infection_monkey/monkey.py

```diff
@@ -519,10 +519,10 @@ def _stop_relay(self):
         if self._relay and self._relay.is_alive():
             self._relay.stop()
 
-            while self._relay.is_alive() and not self._control_channel.should_agent_stop():
+            while self._relay.is_alive() and not self._agent_lifecycle.should_agent_stop():
                 self._relay.join(timeout=5)
 
-            if self._control_channel.should_agent_stop():
+            if self._agent_lifecycle.should_agent_stop():
                 self._relay.join(timeout=60)
```

Comment:
I've spotted a bug. It's possible that the agent is shutting down because communication with the Island has been disrupted. This means `should_agent_shop()` could raise an exception, causing a number of cleanup operations in `cleanup()` to be skipped.

Reply:
@ilija-lazoroski Fixed. Take a look. We should probably run and observe tunneling tests, but I think the change are fairly straight forward.